### DISCUSSION TITLE
Multiple keys

### DISCUSSION THREAD

**Discussion by Daoudou** [openscenegraph/OpenSceneGraph#1339] [Q&A]
Hello 

i want to know how can i  get multiple key, for example : 
hold ctrl key and a pressed : ctrkl + a

do you think i'm in the good way ? 

`bool KeyEventCamera::handle(const osgGA::GUIEventAdapter& ea, osgGA::GUIActionAdapter& aa)
{
#define PRINT(mask) osg::notify(osg::NOTICE)<<#mask<<" ="<<(ea.getModKeyMask() & mask)<<std::endl;
	
	switch (ea.getEventType())
	{
	case osgGA::GUIEventAdapter::KEYDOWN:
		if (ea.getModKeyMask() && osgGA::GUIEventAdapter::MODKEY_CTRL)
		{
			if (ea.getKey() == 'a')
			{
				std::cout << "A key pressed" << "\n";
			}
		}
		break;
	default:
		break;
	}
	return false;
}
`

thanks in advance ^^

**Comment by xarray** [comment]
As far as I know, This may not work for all platforms. Especially, Ctrl+A will generate a different key value under Windows. You may have to capture that special key value instead.

**Comment by robertosfield** [comment]
Try using ea.getUnmodifiedKey() rather than ea.getKey().

**Reply by Daoudou** [reply]
Thanks .

IT'S WORKS 🤩

if anyone needs it

`bool KeyEventCamera::handle(const osgGA::GUIEventAdapter& ea, 
	osgGA::GUIActionAdapter& aa)
{
	if (!_trackballManipulator) return false;
	osg::Matrix matrix;

	switch (ea.getEventType())
	{
	case osgGA::GUIEventAdapter::KEYDOWN:
		if (ea.getModKeyMask() &
			osgGA::GUIEventAdapter::MODKEY_CTRL)
		{
			if (ea.getUnmodifiedKey() == osgGA::GUIEventAdapter::KEY_S) 
			{
				std::cout << "ctrl + s keys pressed" << "\n";
			}
		}
		break;
	default:
		break;
	}

	return false;
}`
